test/test_quantumchannel.jl

```diff
@@ -0,0 +1,34 @@
+using QuantumSavory
+using ResumableFunctions
+using ConcurrentSim
+using Test
+
+bell = (Z1⊗Z1 + Z2⊗Z2)/sqrt(2.0)
+regA = Register(2)
+regB = Register(2)
+initialize!((regA[1], regB[2]), bell)
+
+sim = Simulation()
+
+# Delay queue for quantum channel 
+queue = DelayQueue{RegRef}(sim, 10.0)
+
+qc = QuantumChannel(queue)
+
+@resumable function alice_node(env, qc)
+    put!(qc, regA[1])
+end
+
+@resumable function bob_node(env, qc)
+    @yield @process take!(env, qc, regB[1])
+end
+
+@process alice_node(sim, qc)
+@process bob_node(sim, qc)
+
+run(sim)
+
+sref = regB.staterefs[1]
+
+# the above code puts both the qubits of the state in the same register
+@test sref.registers[1] == sref.registers[2]
```

Comment:
we should have a few tests that double-check that the time evolution due to backgrounds is properly evaluated, i.e. that all the necessary `uptotime!` are called internally.